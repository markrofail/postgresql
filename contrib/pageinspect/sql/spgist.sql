CREATE TABLE spgist_table (t box);
\copy spgist_table from 'data/rect.data'

SELECT center(t) AS p INTO spgist_table_p FROM spgist_table;

CREATE INDEX spgist_idx ON spgist_table_p USING spgist (p);
CREATE INDEX kdspgist_idx ON spgist_table_p USING spgist (p kd_point_ops);

SELECT spgist_stats('spgist_idx');
SELECT * FROM spgist_print('kdspgist_idx') as t(tid tid, allthesame bool, node_n int, level int, tid_pointer tid, prefix float8, node_label int, leaf_value point);
SELECT * FROM spgist_print('spgist_idx') as t(tid tid, allthesame bool, node_n int, level int, tid_pointer tid, prefix point, node_label int, leaf_value point) WHERE level = 1;
