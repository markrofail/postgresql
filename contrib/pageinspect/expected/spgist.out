CREATE TABLE spgist_table (t box);
\copy spgist_table from 'data/rect.data'
SELECT center(t) AS p INTO spgist_table_p FROM spgist_table;
CREATE INDEX spgist_idx ON spgist_table_p USING spgist (p);
CREATE INDEX kdspgist_idx ON spgist_table_p USING spgist (p kd_point_ops);
SELECT spgist_stats('spgist_idx');
           spgist_stats           
----------------------------------
 totalPages:        21           +
 deletedPages:      0            +
 innerPages:        2            +
 leafPages:         19           +
 emptyPages:        0            +
 usedSpace:         126.70 kbytes+
 usedInnerSpace:    1.17 kbytes  +
 usedLeafSpace:     125.53 kbytes+
 freeSpace:         40.64 kbytes +
 fillRatio:         75.71%       +
 leafTuples:        3947         +
 innerTuples:       20           +
 innerAllTheSame:   0            +
 leafPlaceholders:  569          +
 innerPlaceholders: 0            +
 leafRedirects:     0            +
 innerRedirects:    0
(1 row)

SELECT * FROM spgist_print('kdspgist_idx') as t(tid tid, allthesame bool, node_n int, level int, tid_pointer tid, prefix float8, node_label int, leaf_value point);
   tid    | allthesame | node_n | level | tid_pointer | prefix  | node_label |    leaf_value     
----------+------------+--------+-------+-------------+---------+------------+-------------------
 (1,1)    | f          |      0 |     1 | (5,1)       |   23030 |            | 
 (1,1)    | f          |      1 |     1 | (5,2)       |   23030 |            | 
 (5,2)    | f          |      0 |     2 | (5,6)       |   21664 |            | 
 (5,2)    | f          |      1 |     2 | (5,3)       |   21664 |            | 
 (5,3)    | f          |      0 |     3 | (5,5)       | 37159.5 |            | 
 (5,3)    | f          |      1 |     3 | (5,12)      | 37159.5 |            | 
 (5,12)   | f          |      0 |     4 | (5,13)      | 36357.5 |            | 
 (5,12)   | f          |      1 |     4 | (14,226)    | 36357.5 |            | 
 (14,226) |            |        |     5 |             |         |            | (43240,47256.5)
 (5,13)   | f          |      0 |     5 | (3,112)     |   43507 |            | 
 (5,13)   | f          |      1 |     5 | (5,25)      |   43507 |            | 
 (5,25)   | f          |      0 |     6 | (22,116)    |   28862 |            | 
 (5,25)   | f          |      1 |     6 | (22,115)    |   28862 |            | 
 (22,115) |            |        |     7 |             |         |            | (30902,44532)
 (22,116) |            |        |     7 |             |         |            | (23048,47416)
 (3,112)  |            |        |     6 |             |         |            | (35218.5,42124)
 (5,5)    | f          |      0 |     4 | (5,11)      |   33416 |            | 
 (5,5)    | f          |      1 |     4 | (5,22)      |   33416 |            | 
 (5,22)   | f          |      0 |     5 | (20,225)    |   30991 |            | 
 (5,22)   | f          |      1 |     5 | (5,23)      |   30991 |            | 
 (5,23)   | f          |      0 |     6 | (9,112)     |   41820 |            | 
 (5,23)   | f          |      1 |     6 | (9,114)     |   41820 |            | 
 (9,114)  |            |        |     7 |             |         |            | (44732,32182)
 (9,112)  |            |        |     7 |             |         |            | (35580,33526.5)
 (20,225) |            |        |     6 |             |         |            | (47724.5,27185.5)
 (5,11)   | f          |      0 |     5 | (13,101)    | 29986.5 |            | 
 (5,11)   | f          |      1 |     5 | (13,100)    | 29986.5 |            | 
 (13,100) |            |        |     6 |             |         |            | (24069,30850.5)
 (13,101) |            |        |     6 |             |         |            | (29539,25566)
 (5,6)    | f          |      0 |     3 | (5,7)       |   36774 |            | 
 (5,6)    | f          |      1 |     3 | (5,19)      |   36774 |            | 
 (5,19)   | f          |      0 |     4 | (5,20)      | 10075.5 |            | 
 (5,19)   | f          |      1 |     4 | (18,225)    | 10075.5 |            | 
 (18,225) |            |        |     5 |             |         |            | (20920.5,49105.5)
 (5,20)   | f          |      0 |     5 | (10,110)    | 44171.5 |            | 
 (5,20)   | f          |      1 |     5 | (10,113)    | 44171.5 |            | 
 (10,113) |            |        |     6 |             |         |            | (93,46797)
 (10,110) |            |        |     6 |             |         |            | (28.5,38640.5)
 (5,7)    | f          |      0 |     4 | (7,113)     |    9517 |            | 
 (5,7)    | f          |      1 |     4 | (5,21)      |    9517 |            | 
 (5,21)   | f          |      0 |     5 | (19,115)    | 28907.5 |            | 
 (5,21)   | f          |      1 |     5 | (19,112)    | 28907.5 |            | 
 (19,112) |            |        |     6 |             |         |            | (11916.5,31668)
 (19,115) |            |        |     6 |             |         |            | (20622.5,27462.5)
 (7,113)  |            |        |     5 |             |         |            | (9296,35157)
 (5,1)    | f          |      0 |     2 | (5,8)       |   26938 |            | 
 (5,1)    | f          |      1 |     2 | (5,4)       |   26938 |            | 
 (5,4)    | f          |      0 |     3 | (5,10)      |    9532 |            | 
 (5,4)    | f          |      1 |     3 | (5,14)      |    9532 |            | 
 (5,14)   | f          |      0 |     4 | (5,15)      |   38603 |            | 
 (5,14)   | f          |      1 |     4 | (15,222)    |   38603 |            | 
 (15,222) |            |        |     5 |             |         |            | (41926.5,17934.5)
 (5,15)   | f          |      0 |     5 | (8,113)     |   16345 |            | 
 (5,15)   | f          |      1 |     5 | (8,112)     |   16345 |            | 
 (8,112)  |            |        |     6 |             |         |            | (32425,20702.5)
 (8,113)  |            |        |     6 |             |         |            | (29134,15559.5)
 (5,10)   | f          |      0 |     4 | (12,94)     | 38800.5 |            | 
 (5,10)   | f          |      1 |     4 | (5,24)      | 38800.5 |            | 
 (5,24)   | f          |      0 |     5 | (21,108)    |    4752 |            | 
 (5,24)   | f          |      1 |     5 | (21,107)    |    4752 |            | 
 (21,107) |            |        |     6 |             |         |            | (49822.5,7097.5)
 (21,108) |            |        |     6 |             |         |            | (40315.5,1689.5)
 (12,94)  |            |        |     5 |             |         |            | (30295.5,5090)
 (5,8)    | f          |      0 |     3 | (5,17)      | 11733.5 |            | 
 (5,8)    | f          |      1 |     3 | (5,9)       | 11733.5 |            | 
 (5,9)    | f          |      0 |     4 | (6,114)     |   11993 |            | 
 (5,9)    | f          |      1 |     4 | (5,16)      |   11993 |            | 
 (5,16)   | f          |      0 |     5 | (16,123)    |   17591 |            | 
 (5,16)   | f          |      1 |     5 | (16,127)    |   17591 |            | 
 (16,127) |            |        |     6 |             |         |            | (18352.5,19366)
 (16,123) |            |        |     6 |             |         |            | (24795,14921)
 (6,114)  |            |        |     5 |             |         |            | (6706,16676)
 (5,17)   | f          |      0 |     4 | (5,18)      | 13329.5 |            | 
 (5,17)   | f          |      1 |     4 | (17,226)    | 13329.5 |            | 
 (17,226) |            |        |     5 |             |         |            | (23690,10214.5)
 (5,18)   | f          |      0 |     5 | (11,113)    |    6375 |            | 
 (5,18)   | f          |      1 |     5 | (11,109)    |    6375 |            | 
 (11,109) |            |        |     6 |             |         |            | (5501.5,9916)
 (11,113) |            |        |     6 |             |         |            | (1072.5,4752)
(79 rows)

SELECT * FROM spgist_print('spgist_idx') as t(tid tid, allthesame bool, node_n int, level int, tid_pointer tid, prefix point, node_label int, leaf_value point) WHERE level = 1;
  tid  | allthesame | node_n | level | tid_pointer |               prefix                | node_label | leaf_value 
-------+------------+--------+-------+-------------+-------------------------------------+------------+------------
 (1,1) | f          |      0 |     1 | (5,4)       | (24530.2070484581,23595.7092511013) |            | 
 (1,1) | f          |      1 |     1 | (5,3)       | (24530.2070484581,23595.7092511013) |            | 
 (1,1) | f          |      2 |     1 | (5,2)       | (24530.2070484581,23595.7092511013) |            | 
 (1,1) | f          |      3 |     1 | (5,1)       | (24530.2070484581,23595.7092511013) |            | 
(4 rows)

